pkg/pgmodel/ingestor/trace/writer.go

```diff
@@ -296,7 +302,7 @@ func (t *traceWriterImpl) InsertTraces(ctx context.Context, traces pdata.Traces)
 				}
 
 				spanRows = append(spanRows, []interface{}{traceID, spanID, parentSpanID,
-					operationID, start, end, getTraceStateValue(span.TraceState()), jsonTags, span.DroppedAttributesCount(), eventTimeRange, span.DroppedEventsCount(),
+					operationID, start, end, float64(end.Sub(start).Nanoseconds()) / float64(1e6), getTraceStateValue(span.TraceState()), jsonTags, span.DroppedAttributesCount(), eventTimeRange, span.DroppedEventsCount(),
```

Comment:
You can also do this if the aim is to covert in milliseconds.

```suggestion
					operationID, start, end, float64(timestamp.FromFloatSeconds(end.Sub(start).Seconds())), getTraceStateValue(span.TraceState()), jsonTags, span.DroppedAttributesCount(), eventTimeRange, span.DroppedEventsCount(),
```